src/QsCompiler/CompilationManager/ProcessingQueue.cs

```diff
@@ -4,142 +4,191 @@
+                try
+                {
+                    QsCompilerError.RaiseOnFailure(action, this.exceptionHeader);
+                }
+                catch (Exception ex)
+                {
+                    this.logException(ex);
+                }
             });
 
         /// <summary>
-        /// The queue will not accept any more tasks after calling complete. 
+        /// The queue will not accept any more tasks after calling complete.
         /// </summary>
         public void Complete() =>
-            this.Scheduler.Complete();
-
+            this.scheduler.Complete();
 
         // non-concurrent routines - i.e. routines that are forced to execute in order
 
         /// <summary>
-        /// Enqueues the given Action for exclusive (serialized) execution. 
-        /// Uses the set exception logger to log any exception that occurs during execution. 
+        /// Enqueues the given Action for exclusive (serialized) execution.
+        /// Uses the set exception logger to log any exception that occurs during execution.
         /// Throws an ArgumentNullException if the given Action is null.
         /// </summary>
         public Task QueueForExecutionAsync(Action processing)
         {
-            if (processing == null) throw new ArgumentNullException(nameof(processing));
-            var processingTask = ProcessingTaskAsync(processing);
-            processingTask.Start(this.Scheduler.ExclusiveScheduler);
+            if (processing == null)
+            {
+                throw new ArgumentNullException(nameof(processing));
+            }
+            var processingTask = this.ProcessingTaskAsync(processing);
+            processingTask.Start(this.scheduler.ExclusiveScheduler);
             return processingTask;
         }
 
         /// <summary>
         /// Executes the given Action synchronously, with no exclusive actions running.
-        /// Uses the set exception logger to log any exception that occurs during execution. 
+        /// Uses the set exception logger to log any exception that occurs during execution.
         /// Throws an ArgumentNullException if the given Action is null.
-        /// NOTE: may deadlock if the given function to execute calls this processing queue. 
+        /// NOTE: may deadlock if the given function to execute calls this processing queue.
         /// </summary>
         public void QueueForExecution(Action processing)
         {
-            if (processing == null) throw new ArgumentNullException(nameof(processing));
-            this.QueueForExecution(() => { processing(); return new object(); }, out var _);
+            if (processing == null)
+            {
+                throw new ArgumentNullException(nameof(processing));
+            }
+            this.QueueForExecution(
+                () =>
+                {
+                    processing();
+                    return new object();
+                }, out _);
```

Comment:
Is this the correct style for lambdas moving forward? Does the `() =>` have to be on its own line? And the last parameter is _not_ on its own line?

Reply:
The lambda arrow is on a new line because of [SA1116: Split Parameters Must Start On Line After Declaration](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1116.md), which seems to be triggered since `QueueForExecution` has more than one argument. If `QueueForExecution` had only one argument, you could write:

```csharp
this.QueueForExecution(() =>
{
    processing();
    return new object();
});
```

I'm not sure why `out _` is allowed to be on the same line as the closing brace. Normally you can't have more than one argument on the same line unless all of them are on the same line. Maybe StyleCop considers the closing brace in `}, out _` to not count as part of the previous argument, but I agree it looks weird.

Reply:
I see. I may try to avoid this in the future by being careful about the parameters I make for functions, because this looks pretty rough to me.

Reply:
If it helps, it is not required to have the argument on the same line as the closing brace. You can also write:

```csharp
this.QueueForExecution(
    () =>
    {
        processing();
        return new object();
    },
    out _);
```

Reply:
That does help a bit, but it is still confusing to have the first argument, the lambda, take up multiple lines on the same indentation level because its argument list is separated from its body. At first glace it almost looks like there are 3 arguments to the method.